runelite-client/src/main/java/net/runelite/client/plugins/barrows/BarrowsPlugin.java

```diff
@@ -221,6 +245,70 @@ else if (event.getGroupId() == WidgetID.BARROWS_PUZZLE_GROUP_ID)
 		}
 	}
 
+	@Subscribe
+	public void onGameTick(GameTick tick)
+	{
+		if (isInCrypt())
+		{
+			// Get new potential after a kill
+			int newPotential = client.getVar(Varbits.BARROWS_REWARD_POTENTIAL);
+			// Calculate the difference between the original potential and new potential (how much potential gained
+			// from the kill)
+			int potentialDifference = newPotential - this.barrowsStartingPotential;
+			// If the amount of potential goes down, a barrows trip has been completed and we want to reset the counters
+			// Note - this only occurs when re-entering a crypt/maze after having completed, so if a player finishes a
+			// barrows run, the kill count will still be present until they leave the crypt.  But on re-entry of a new
+			// crypt, it will have reset.
+			if (potentialDifference < 0)
+			{
+				initializeKC();
+			}
+			// Set the "original" potential to the new potential so future calculations are based on the correct value.
+			this.barrowsStartingPotential = newPotential;
+
+			for (CryptMonsters monster : CryptMonsters.values())
+			{
+				// To figure out which monster was killed, just check the difference in reward potential and compare it
+				// to the combat level of the monster.  This does mean, however, that if multiple unregistered kills occur
+				// (such as if the tracker is toggled off mid-run, then back on after a few kills), they will not appear.
+				if (monster.getCombatLevel() == potentialDifference)
+				{
+					String monsterSlain = monster.getName();
+					int numKilled = monstersKilled.get(monsterSlain);
+					monstersKilled.put(monsterSlain, numKilled + 1);
+					//Update the tooltip
+					this.kcInfobox.updateTooltip();
+				}
```

Comment:
This should `break;` to exit the loop once matched.
```suggestion
				if (monster.getCombatLevel() == potentialDifference)
				{
					String monsterSlain = monster.getName();
					int numKilled = monstersKilled.get(monsterSlain);
					monstersKilled.put(monsterSlain, numKilled + 1);

					//Update the tooltip
					this.kcInfobox.updateTooltip();

					break;
				}
```